scripts/ci/sync_extensions.py

```diff
@@ -0,0 +1,141 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=broad-except
+
+import os
+import re
+import json
+from subprocess import check_output
+
+DEFAULT_TARGET_INDEX_URL = os.getenv('AZURE_EXTENSION_TARGET_INDEX_URL')
+STORAGE_ACCOUNT_KEY = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT_KEY')
+STORAGE_ACCOUNT = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT')
+STORAGE_CONTAINER = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_CONTAINER')
+
+
+def _get_updated_extension_names():
+    cmd = 'git --no-pager diff --diff-filter=ACMRT HEAD~1 -- src/index.json'
```

Comment:
I'm still little concern on replying on change on index.json. what if the pipeline run failed? new run need to catch up extension blob changes in last several index.json change.

Reply:
Added environment variable to control commit numbers to diff. So in case of failure, we can manually trigger the pipeline and pick up all the changes for the last n commits.